src/snowflake/snowpark/_internal/analyzer/unary_plan_node.py

```diff
@@ -89,18 +188,50 @@ def __init__(
         super().__init__(child)
         self.column_map = column_map
 
+    @property
+    def individual_node_complexity(self) -> Dict[str, int]:
+        # SELECT * RENAME (before AS after, ...) FROM child
+        return {
+            PlanNodeCategory.COLUMN.value: 1 + 2 * len(self.column_map),
```

Comment:
why it has a 2* here? shouldn't before as after counted as 1 column?

Reply:
for each col1 AS col2, we are counting complexity as col1.complexity + col2.complexity. Which makes 2* len(column_map) in this case